source/includes/write/gridfs.php

```diff
@@ -0,0 +1,89 @@
+<?php
+require 'vendor/autoload.php'; // include Composer's autoloader
+
+use MongoDB\Client;
+use MongoDB\BSON\ObjectID;
+use MongoDB\GridFS\Bucket;
+use MongoDB\Driver\Manager;
+
+$uri = getenv('MONGODB_URI') ?: throw new RuntimeException('Set the MONGODB_URI variable to your Atlas URI that connects to the sample dataset');
+$client = new MongoDB\Client($uri);
+
+// start-to-json
+function toJSON(object $document): string
+{
+    return MongoDB\BSON\Document::fromPHP($document)->toRelaxedExtendedJSON();
+}
+// end-to-json
+
+// Creates a GridFS bucket or references an existing one
+// start-create-bucket
+$bucket = $client->db->selectGridFSBucket();
+// end-create-bucket
+
+// Creates or references a GridFS bucket with a custom name
+// start-create-custom-bucket
+$custom_bucket = $client->db->selectGridFSBucket(
+    ['bucketName' => 'myCustomBucket']
+);
+// end-create-custom-bucket
+
+// Uploads a file called "my_file" to the GridFS bucket and writes data to it
+// start-open-upload-stream
+$stream = $bucket->openUploadStream('my_file', [
+    'metadata' => ['contentType' => 'text/plain']
+]);
+fwrite($stream, 'Data to store');
+fclose($stream);
+// end-open-upload-stream
+
+// Uploads data to a stream, then writes the stream to a GridFS file
+// start-upload-from-stream
+$stream = fopen('php://temp', 'w+b');
+fwrite($stream, 'Data to store');
+rewind($stream);
+
+$bucket->uploadFromStream('new_file', $stream);
+// end-upload-from-stream
+
+// Prints information about each file in the bucket
+// start-retrieve-file-info
+$files = $bucket->find();
+foreach ($files as $file_doc) {
+    echo toJSON($file_doc), PHP_EOL;
+}
+// end-retrieve-file-info
+
+// Downloads the "my_file" file from the GridFS bucket and prints its contents
+// start-open-download-stream-name
+$stream = $bucket->openDownloadStreamByName('my_file');
+$contents = stream_get_contents($stream);
+echo $contents, PHP_EOL;
+fclose($stream);
+// end-open-download-stream-name
+
+// Downloads a file from the GridFS bucket by referencing its ObjectID value
+// start-download-files-id
+$stream = $bucket->openDownloadStream(new ObjectID('66e0a5487c880f844c0a32b1'));
+$contents = stream_get_contents($stream);
+fclose($stream);
+// end-download-files-id
+
+// Downloads an entire GridFS file to a download stream
+// start-download-to-stream
+$stream = fopen('php://temp', 'w+b');
```

Comment:
I would also suggest using a local filename here, as we did in the [original tutorial](https://www.mongodb.com/docs/php-library/current/tutorial/gridfs/#downloading-files-with-readable-streams). Experienced users will know they can use any type of stream here (including [PHP's special streams](https://www.php.net/manual/en/wrappers.php.php) like `php://temp`), but most readers would benefit from seeing how to efficiently copy a file out of GridFS.

The hasty, inefficient solution would be for users to rely on `openDownloadStream()`, `stream_get_contents()`, and [`file_put_contents()`](https://www.php.net/manual/en/function.file-put-contents.php), which has the same problem as the `file_get_contents()` approach I discussed for uploads.

FYI the `downloadToStream()` family of functions is also a [thin wrapper](https://github.com/mongodb/mongo-php-library/blob/1.19.1/src/GridFS/Bucket.php#L271) around `stream_copy_to_stream()`, like `Bucket::uploadFromStream()`.